### ISSUE TITLE
Sequence | Hashable reduced to Hashable when class is Generic

### ISSUE THREAD

**Issue by twoertwein** [python/mypy#14311]
**Bug Report**

**To Reproduce**

```python
from typing import (
    Generic,
    Hashable,
    Sequence,
    TypeVar,
)

T = TypeVar("T")

class Series(Generic[T]):
    def reset_index(self, level: Sequence | Hashable = ...): ...

Series().reset_index(["ab"])  # fails
reveal_type(
    Series().reset_index
)  # Revealed type is "def (level: typing.Hashable =) -> Any"

class Series2:  # not generic
    def reset_index(self, level: Sequence | Hashable = ...): ...

Series2().reset_index(["ab"])  # works
reveal_type(
    Series2().reset_index
)  # Revealed type is "def (level: Union[typing.Sequence[Any], typing.Hashable] =) -> Any"```
```

**Expected Behavior**

`Series().reset_index(["ab"]) ` is accepted since `list[str]` is a `Sequence`.

**Actual Behavior**

`Sequence | Hashable` is reduced to just `Hashable` which is incompatible with `list[str]`.

**Your Environment**

- Mypy version used: 0.991
- Python version used: 3.11.0

**Comment by erictraut** [comment]
This appears to have been fixed in mypy 1.0.0.

**Comment by hauntsaninja** [comment]
Fixed in #14178
